sdk/storage/storage-blob/src/BlobBatchClient.ts

```diff
@@ -63,8 +65,20 @@ export declare type BlobBatchSetBlobsAccessTierResponse = BlobBatchSubmitBatchRe
 export class BlobBatchClient {
   private _serviceContext: Service;
 
-  constructor(service: Service) {
-    this._serviceContext = service;
+  /**
+   * Creates an instance of BlobBatchClient.
+   *
+   * @param {string} url A Client string pointing to Azure Storage blob service, such as
+   *                     "https://myaccount.blob.core.windows.net". You can append a SAS
+   *                     if using AnonymousCredential, such as "https://myaccount.blob.core.windows.net?sasString".
+   * @param {Pipeline} pipeline Call newPipeline() to create a default
+   *                            pipeline, or provide a customized pipeline.
+   * @memberof BlobServiceClient
+   */
+  constructor(url: string, pipeline: Pipeline) {
```

Comment:
Minor: Please ensure that's the latest design across langs. You know another possible ctor, is to accept ServiceClient as param.

Reply:
Why is the pipeline mandatory here?

Reply:
We can make Pipeline optional.

The two reasons I chose not to accept an instance of ServiceClient are:

1) It would create another circular reference and we already are facing a webpack bug due to these.
2) We wouldn't be able to access ServiceClient's storageContext or pipeline without an `any` cast to access these private members, which is brittle.

Reply:
To circle back on this, I'd like to see if users give us feedback that they miss being able to pass in ServiceClient (even though they can retrieve an instance of BatchClient directly from it.) 

If so, I think we can look at making pipeline public without it hurting anything, which is what .NET does: https://github.com/Azure/azure-sdk-for-net/blob/7c2ff45b0f4c60fc9ed2420969b556851e74cd4d/sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatchClient.cs#L67

I'd prefer to have a solution for the circular reference issue before we tackle that though.